src/ConstructionBase.jl

```diff
@@ -55,7 +57,40 @@ function setproperties(obj; kw...)
 end
 
 setproperties(obj, patch::NamedTuple) = _setproperties(obj, patch)
+setproperties(obj::Tuple, patch::typeof(NamedTuple())) = obj
+function setproperties(obj::Tuple, patch::NamedTuple)
+    msg = """
+    Tuple has no named properties.
+    obj  ::Tuple      = $obj
+    patch::NamedTuple = $patch
+    """
+    throw(ArgumentError(msg))
+end
 
+function setproperties(obj::Tuple, patch::Tuple)
+    setproperties_tuple(obj, patch)
+end
+function setproperties_tuple(obj::NTuple{N,Any}, patch::NTuple{N,Any}) where {N}
+    patch
+end
+append(x,y) = (x..., y...)
+function setproperties_tuple(obj::NTuple{N,Any}, patch::NTuple{K,Any}) where {N,K}
+    if K > N
```

Comment:
I can do that, but why? Do you think this extra amount of code will influence inline heuristics or something?

Reply:
Ok so you think there could be again string interpolation allocation issues here right?

Reply:
I have no idea honestly I just do this blindly following tradition others have laid down that is probably out of data. I remember seeing it have some affects a few years ago somewhere.

Reply:
Yeah same thing for me I also have my collection of weird performance rituals I don't really understand or know if they are outdated. I switched to avoid these and only try them if I have an actual example of bad performance. My hope is I can learn a bit better which patterns are superstitions and which are not. I must admit I learned very little this way though.